src/ApiCore/Parser.php

```diff
@@ -60,29 +60,38 @@ public function __construct()
      */
     public function parse($data)
     {
+        if (is_null($data)) {
```

Comment:
Does [empty()](http://php.net/manual/en/function.empty.php) (available starting from php4) allow to combine these two if statements in one? Youl would have to have one error message instead of two more specific ones (something like `'Cannot parse empty $data parameter'`, but it seems ok for this case). Same applies to similar check in `PathTemplate.__constructor()`.

Reply:
Fixed